lmdeploy/pytorch/nn/rotary_embedding.py

```diff
@@ -1,12 +1,91 @@
 # Copyright (c) OpenMMLab. All rights reserved.
 from torch import Tensor, nn
+from transformers import PretrainedConfig
 
 from ..backends import OpType, get_backend
 from ..backends.rotary_embedding import (Llama3Parameters,
                                          LongRoPEScalingParameters, RopeType,
                                          YarnParameters)
 
 
+def _get_default_rope_parameters(config: PretrainedConfig):
+    """get default rope parameters."""
+    return dict(emb_type=RopeType.Default, scaling_factor=1.0)
+
+
+def _get_linear_scaling_rope_parameters(config: PretrainedConfig):
+    """get linear rope parameters."""
+    rope_scaling = config.rope_scaling
+    scaling_factor = rope_scaling['factor']
+    return dict(emb_type=RopeType.LinearScaling, scaling_factor=scaling_factor)
+
+
+def _get_dynamic_ntk_parameters(config: PretrainedConfig):
+    """get dynamic ntk parameters."""
+    rope_scaling = config.rope_scaling
+    scaling_factor = rope_scaling['factor']
+    return dict(emb_type=RopeType.DynamicNTKScaling,
+                scaling_factor=scaling_factor)
+
+
+def _get_yarn_parameters(config: PretrainedConfig):
+    """get yarn parameters."""
+    rope_scaling = config.rope_scaling
+    scaling_factor = rope_scaling['factor']
```

Comment:
This is not general enough, some model might name it 'scaling_factor' or something else.

Reply:
I followed [transformers](https://github.com/huggingface/transformers/blob/5f4ee98a7ade33e1c54fdd6181d04ee7b426b392/src/transformers/modeling_rope_utils.py#L403)(ROPE_INIT_FUNCTIONS) which should suit for builtin models.

Maybe custom models like [deepseek](https://github.com/InternLM/lmdeploy/blob/main/lmdeploy/pytorch/models/deepseek_v2.py#L548) should not use `build_rotary_params` and build rotary_params in a custom way.